### PR TITLE
Migrate integration tests over to aws-go-sdk-v2

### PR DESCRIPTION
**Pull request by Skarlso**
### Description

Closes https://github.com/weaveworks/eksctl/issues/5099

_NOTE_: This is now also degrading aws-iam-authenticator to 0.5.5 in main. Incidentally, that's actually the version that we are using in the Build already! So we should align that! :)

### Checklist
- [ ] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [ ] Manually tested
- [ ] Made sure the title of the PR is a good description that can go into the release notes
- [ ] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

File: integration/matchers/matchers.go [hunk 1]
```diff
@@ -73,28 +71,28 @@ func (m *existingCluster) Match(actual interface{}) (success bool, err error) {
...
 		Name: aws.String(m.expectedName),
 	}
-	output, err := eks.DescribeCluster(input)
+	output, err := client.DescribeCluster(context.Background(), input)
 
 	if err != nil {
-		// Check if its a not found error: ResourceNotFoundException
-		if !strings.Contains(err.Error(), awseks.ErrCodeResourceNotFoundException) {
+		// Check if it's a not found error: ResourceNotFoundException
+		if !strings.Contains(err.Error(), "ResourceNotFoundException") {
```
Comment (cPu1) [inline integration/matchers/matchers.go:87]: A more idiomatic approach (requires importing `aws-sdk-go-v2/service/eks/types`):

```suggestion
		var notFoundErr *ekstypes.ResourceNotFoundException
		if !errors.As(err, &notFoundErr) {
```

File: integration/tests/identity_provider/identity_provider_test.go [hunk 2]
```diff
@@ -125,7 +121,9 @@ var _ = Describe("(Integration) [Identity Provider]", func() {
 		test.CreateClusterRoleBindingFromFile("testdata/cluster-role-binding.yaml")
 
 		By("creating an OIDC Clientset")
-		clientset, err := createOIDCClientset(eks.New(NewSession(params.Region)), oidcConfig, params.ClusterName)
+		e := eks.NewFromConfig(NewConfig(params.Region))
+		//clientset, err := createOIDCClientset(e, oidcConfig, params.ClusterName)
```
Comment (cPu1) [inline integration/tests/identity_provider/identity_provider_test.go:125]:
```suggestion
```

**Review by cPu1 — APPROVED** [review]
LGTM.

File: integration/matchers/matchers.go (hunk 1 above)
Comment (Skarlso) [reply integration/matchers/matchers.go:87]: Yeah I was lazy hahahahhaa, thanks. Also I need to run integration tests on this branch. :D

**Comment by Skarlso** [comment]
New test run with the fixes: https://github.com/weaveworks/eksctl-ci/actions/runs/2311987986

**Comment by Skarlso** [comment]
Another test run fixing a different matcher... **sigh**

https://github.com/weaveworks/eksctl-ci/actions/runs/2312810096

**Review by Himangini — APPROVED** [review]
👍🏻

**Pull request merged**